core/src/main/java/io/grpc/util/RoundRobinLoadBalancer.java

```diff
@@ -296,13 +298,7 @@ public String toString() {
     }
 
     private Subchannel nextSubchannel() {
-      int size = list.size();
-      int i = indexUpdater.incrementAndGet(this);
-      if (i >= size) {
-        int oldi = i;
-        i %= size;
-        indexUpdater.compareAndSet(this, oldi, i);
-      }
+      int i = indexUpdater.updateAndGet(this, old -> old == maxIndex ? 0 : old + 1);
```

Comment:
Our minimum supported Java version is 7 (although we are about to drop Java 7 support in the short future), so please avoid using lambda.